crates/ruff_db/src/diagnostic/render.rs

```diff
@@ -815,6 +814,50 @@ watermelon
         );
     }
 
+    #[test]
+    fn no_range() {
+        let mut env = TestEnvironment::new();
+        env.add("animals", ANIMALS);
+
+        let mut builder = env.err();
+        builder
+            .diag
+            .annotate(Annotation::primary(builder.env.path("animals")));
+        let diag = builder.build();
+        insta::assert_snapshot!(
+            env.render(&diag),
+            @r"
+        error: lint:test-diagnostic: main diagnostic message
+         --> /animals:1:1
+          |
+        1 | aardvark
+          | ^
+        2 | beetle
+        3 | canary
+          |
+        ",
+        );
```

Comment:
I'm torn on this. Mainly because it will break https://github.com/astral-sh/ruff/pull/17631 😅 

The reason I want to include a `File` in that message is so that the diagnostic sorting keeps working. But I don't think I want a code frame. 

I think I'd be fine if we don't render primary annotations without a span or message.

Reply:
You know what. Let me change my PR. The sorting doesn't matter in that case because there won't be any other diagnostics for the same file.

Reply:
I lean towards going with this PR and removing the `file` in your other PR. I realize the sorting may not be ideal, but I think it's worse to silently drop some part of the diagnostic from the rendered output.

An alternative for your conundrum is to augment the diagnostic model to support attaching files (or even `Span`s) that aren't annotations/code-frames. But I haven't fully thought that through.